app/code/Magento/Ui/view/base/web/js/modal/prompt.js

```diff
@@ -71,10 +72,15 @@ define([
          * Create widget.
          */
         _create: function () {
+            var modalContentElem;
+            
             this.options.focus = this.options.promptField;
             this.options.validation = this.options.validation && this.options.validationRules.length;
             this._super();
-            this.modal.find(this.options.modalContent).append(this.getFormTemplate());
+            modalContentElem = this.modal.find(this.options.modalContent).append(this.getFormTemplate());
+            if(this.options.formSelector) {
+                modalContentElem.find(this.options.formSelector).on('submit', _.bind(this.submitForm, this));
```

Comment:
Base `modal` widget which this `prompt` is extending has this custom handler option that can be provided for closing the modal https://github.com/magento/magento2/blob/e6531ff15903b00c9e46712711572009cf28dbc8/app/code/Magento/Ui/view/base/web/js/modal/modal.js#L136. Maybe we could allow people to pass custom form submition handler just to be on the safe side?

Reply:
I do not see a benefit - and can be actually confusing. This form is not being submitted by the OK button. Submit here is a side effect. 
I would not add additional possibilities to a "corner case" not predicted by the original functionality of the OK button.